test/unit/services/create_edition_link_monitor_test.rb

```diff
@@ -0,0 +1,27 @@
+require 'test_helper'
+
+class CreateEditionLinkMonitorTest < ActiveSupport::TestCase
+  setup do
+    Whitehall.link_checker_api_client.stubs(:upsert_resource_monitor).returns {}
+  end
+
+  def expect_link_checker_api_call
+    Whitehall.link_checker_api_client.expects(:upsert_resource_monitor)
+      .with(any_parameters)
+  end
```

Comment:
Would it be worth asserting the arguments to this call to ensure the links have been extracted correctly?

Reply:
I wonder if it'd be worth pushing some of these assertion-y things into [gds-api-adapters](https://github.com/alphagov/gds-api-adapters/blob/master/lib/gds_api/test_helpers/link_checker_api.rb)? [This](https://github.com/alphagov/gds-api-adapters/blob/master/lib/gds_api/test_helpers/publishing_api_v2.rb#L167-L170) is an example of an `assert` method in this style.

The benefit(?) is that it tests the assertion at the HTTP layer rather than on the client, so it tests the integration a little bit past the client boundary, rather than at it. It also promotes re-use across the code base / other code bases. What are your thoughts?